src/services/storage/useStorage.ts

```diff
@@ -0,0 +1,53 @@
+import { Storage, Unsubscribe } from '@/services/storage/Storage'
+import { StorageItem } from '@/services/storage/StorageItem'
+import { tryOnScopeDispose } from '@/utilities/tryOnScopeDispose'
+
+export type UseStorage<T extends StorageItem> = {
+  add: (value: T) => void,
+  addAll: (value: T[]) => void,
+  remove: (id: string) => void,
+  removeAll: (ids: string[]) => void,
+  get: (id: string) => T | undefined,
+  getAll: (ids: string[]) => T[],
+}
+
+export function useStorage<T extends StorageItem>(storage: Storage<T>): () => UseStorage<T> {
+  return () => {
+
+    let getUnsubscribe: Unsubscribe | undefined = undefined
+    let getAllUnsubscribe: Unsubscribe | undefined = undefined
+
+    tryOnScopeDispose(() => {
+      getUnsubscribe?.()
+      getAllUnsubscribe?.()
+    })
+
+    const get = (id: string): T | undefined => {
+      const unsubscribe = getUnsubscribe
+
+      getUnsubscribe = storage.subscribe(id)
+      unsubscribe?.()
+
+      return storage.get(id)
+    }
+
+    const getAll = (ids: string[]): T[] => {
+      const unsubscribe = getAllUnsubscribe
+      const unsubscribes = ids.map<Unsubscribe>(id => storage.subscribe(id))
+
+      getAllUnsubscribe = () => unsubscribes.forEach(unsubscribe => unsubscribe())
+      unsubscribe?.()
+
+      return storage.getAll(ids)
+    }
```

Comment:
devx 🪄 

each time you call get it will automatically unsubscribe from the previous value. This is done in a safe way so if you call this multiple times with the same value its fine but if you call it with different values it will remove the previous value from storage if necessary.